### PR TITLE
MB-9244 Include pending SIT extension requests in visual tag counts (office app)

### PR DESCRIPTION
**Pull request by Ryan-Koch**
## Description

This PR seeks to ensure that pending SIT extensions are included in the tag counts for the Move Task Order navigation tab and the shipment left navigation tab (on the move task order page). If there is a pending SIT extension on a shipment it should add 1 to the visual tag count.

This is a smaller change than it looks like. I moved a whole lot of JSON from one file to another in this [commit](https://github.com/transcom/mymove/pull/7443/commits/1e9adc1357e661627c2ab79a67aed20f21a5e656), which is ballooning the lines changed number quite a bit.

## Setup

There's an existing move with `PENDNG` as the code which should work for checking this behavior.

```sh
make db_dev_e2e_populate
make server_run
make office_client_run
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## Screenshots
image

### CODE REVIEW

**Comment by Ryan-Koch** [comment]
This is a smaller change than it looks like. I moved a whole lot of JSON from one file to another in [this commit](https://github.com/transcom/mymove/pull/7443/commits/1e9adc1357e661627c2ab79a67aed20f21a5e656), which is ballooning the lines changed number quite a bit.

File: src/pages/Office/MoveTaskOrder/MoveTaskOrder.test.jsx [hunk 1]
```diff
@@ -1942,4 +651,29 @@ describe('MoveTaskOrder', () => {
...
+      expect(setUnapprovedSITExtensionCount).toHaveBeenCalledWith(1);
+    });
+
+    it('renders the left nav with tag for SIT extension request', () => {
+      expect(wrapper.find('LeftNav').exists()).toBe(true);
+      const navLinks = wrapper.find('LeftNav a');
+      expect(navLinks.at(0).contains('HHG shipment')).toBe(true);
+      expect(navLinks.at(0).contains('1'));
+    });
+  });
```
Comment (duncan-truss) [inline src/pages/Office/MoveTaskOrder/MoveTaskOrder.test.jsx:678]: I know it requires mocking even more data but could we add a test here to make sure it's zero when a SIT extension exists but is in the APPROVED status?

File: src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx [hunk 2]
```diff
@@ -453,8 +478,13 @@ export const MoveTaskOrder = ({ match, ...props }) => {
...
               <a key={`sidenav_${s.id}`} href={`#s-${s.id}`} className={classes}>
                 {s.label}{' '}
-                {unapprovedServiceItemsForShipment[`${s.id}`] > 0 && (
-                  <Tag>{unapprovedServiceItemsForShipment[`${s.id}`]}</Tag>
+                {(unapprovedServiceItemsForShipment[`${s.id}`] > 0 ||
+                  unapprovedSITExtensionForShipment[`${s.id}`] > 0) && (
+                  <Tag>
+                    {(unapprovedServiceItemsForShipment[`${s.id}`] > 0
+                      ? unapprovedServiceItemsForShipment[`${s.id}`]
+                      : 0) + (unapprovedSITExtensionForShipment[`${s.id}`] > 0 ? 1 : 0)}
```
Comment (duncan-truss) [inline src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx:486]: Can this be simplified to:

```javascript
{(unapprovedServiceItemsForShipment[`${s.id}`] || unapprovedSITExtensionForShipment[`${s.id}`]) && (
{(unapprovedServiceItemsForShipment[`${s.id}`] || 0) + (unapprovedSITExtensionForShipment[`${s.id}`] || 0)}
```

**Review by duncan-truss — APPROVED** [review]
LGTM 👍
